src/main/java/org/soen343/services/modules/SHCModule.java

```diff
@@ -22,13 +25,21 @@ public static SHCModule getInstance() {
     /**
      * Update window state
      *
-     * @param windows
+     * @param windows Set of selected Window Id's
      */
     public void updateWindowState(HashSet<Integer> windows) {
+        HashSet<Integer> validWindowIds = new HashSet<>();
 
         //Check permissions:
+        Window w = new Window(0);
```

Comment:
Why do we need to create a window object to validate, can this be done in a rule object?

Reply:
validate is a function implemented by the Window class, I needed a window instance. Not sure what a rule object is ?